### PR TITLE
Changes in ZoomView and Added MagnifierView.

### PR DESCRIPTION
**Pull request by zakisk**

### CODE REVIEW

File: app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt [hunk 1]
```diff
@@ -0,0 +1,137 @@
+package com.jetpack.compose.learning.magnifierview
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.MotionEvent
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.*
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt:8]: Please remove wildcard imports and use full qualified imports as per general practice. You can take a reference from ZoomViewActivity.kt class.

File: app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt:11]: Please remove wildcard imports and use full qualified imports as per general practice. You can take a reference from ZoomViewActivity.kt class.

File: app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import com.jetpack.compose.learning.theme.SystemUiController
+
+
+@SuppressLint("UnusedMaterialScaffoldPaddingParameter")
+class MagnifierViewActivity : AppCompatActivity() {
+
+    companion object {
+        const val MAGNIFIER_ZOOM_FACTOR = 2F
+    }
+    private val Offset.Companion.default
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt:39]: Please add new line between companion object and variable declarations.

File: app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                MagnifierScreen()
+            }
+        }
+    }
+
+
+
+    @OptIn(ExperimentalFoundationApi::class, ExperimentalComposeUiApi::class)
+    @Composable
+    fun MagnifierScreen() {
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt:72]: Kindly rename this function to "MagnifierViewComposable". It's been a good practice to use "Composable" postfix to any Composable functions.

File: app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+        var centerOffset by remember { mutableStateOf(Offset.default) }
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt:74]: Please mention the variable purpose and its usage.

**Comment by yashwantgowla-simform** [comment]
Adding to this, the magnifierView.gif seems stretched out and blurred so please look into it as well.

**Comment by yashwantgowla-simform** [comment]
@zakisk Thanks for showing your interest to make our repo better. Please resolve the comments and we are good to go with your PR.

**Comment by zakisk** [comment]
Thank you for your gentle suggestions I will endeavor to make it concise and useful.

File: app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                    })
+                }
+            ) {
+                MagnifierViewComposable()
+            }
+        }
+    }
+
+
+
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt:92]: Please remove the unnecessary empty line.

File: app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.DpSize
+import androidx.compose.ui.unit.dp
+import com.jetpack.compose.learning.R
+import com.jetpack.compose.learning.theme.AppThemeState
+import com.jetpack.compose.learning.theme.BaseView
+import com.jetpack.compose.learning.theme.SystemUiController
+
+
```
Comment (yashwantgowla-simform) [inline app/src/main/java/com/jetpack/compose/learning/magnifierview/MagnifierViewActivity.kt:49]: Please remove the unnecessary empty line.

**Comment by yashwantgowla-simform** [comment]
@zakisk Thanks for the quick updates. 

1. Please address few comments.
2. Also the gif seems not correct as per the other gif resolutions and please include the system navigation bar on your emulator when you create a gif. 
3. Also if possible, it would be more better if you squash your commits into single one. Currently, it seems 3 commits in this PR, it should be single commit.

**Comment by yashwantgowla-simform** [comment]
Thanks a lot @zakisk for your contribution. LGTM.

**Review by yashwantgowla-simform — APPROVED** [review]

**Pull request merged**